test/realtime/resume.test.js

```diff
@@ -629,5 +624,63 @@ define(['shared_helper', 'async', 'chai'], function (helper, async, chai) {
         closeAndFinish(done, [sender_realtime, receiver_realtime, resumed_receiver_realtime], err);
       }
     });
+
+    it('recover multiple channels', async function () {
+      const NUM_MSGS = 5;
+
+      const txRest = helper.AblyRest();
+      const rxRealtime = helper.AblyRealtime({ transports: [helper.bestTransport] }, true);
+
+      const channelNames = Array(5).fill().map(() => String(Math.random()));
+      const rxChannels = channelNames.map((name) => rxRealtime.channels.get(name));
+
+      await Promise.all(rxChannels.map((channel) => channel.attach()));
+
+      // Do a few publishes on each channel so the channelSerial is set.
+      await Promise.all(channelNames.map(async (name) => {
+        const tx = txRest.channels.get(name);
+        const rx = rxRealtime.channels.get(name);
+        await rx.attach();
+
+        for (let i = 0; i < NUM_MSGS; i++) {
+          const pSubscribe = rx.subscriptions.once();
+          await tx.publish(null, null);
+          await pSubscribe;
+        }
+      }));
+
+      // Get the recovery key before becoming suspended as it will be reset.
+      const recoveryKey = rxRealtime.connection.getRecoveryKey();
+
+      await new Promise((resolve) => helper.becomeSuspended(rxRealtime, resolve));
+
+      // Send some messages after we've detached that should be recovered.
+      for (const name of channelNames) {
+        const tx = txRest.channels.get(name);
+        for (let i = 0; i < NUM_MSGS; i++) {
+          await tx.publish('sentWhileDisconnected', null);
+        }
+      }
+
+      const rxRealtimeRecover = helper.AblyRealtime({ recover: recoveryKey });
+      const rxRecoverChannels = channelNames.map((name) => rxRealtimeRecover.channels.get(name));
+
+      // Attach recovered channels and check we receive the expected messages.
+      await Promise.all(rxRecoverChannels.map(async (channel) => {
+        await channel.attach();
+
+        let recoveredCount = 0;
+        channel.subscribe((msg) => {
+          recoveredCount++;
+        });
+        // Use a timeout to check we don't receive more messages than we expect.
+        await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Can we try avoid using timeouts, I had a chat with @SimonWoolf, seems we are not using timeouts anywhere else in tests

Reply:
That was just copying the existing tests in `resume.test.js` - though i agree kinda messy - have updated